services/galley/src/Galley/API/LegalHold.hs

```diff
@@ -116,26 +116,26 @@ removeSettings zusr tid (RemoveLegalHoldSettingsRequest mPassword) = do
   --     . Log.field "action" (Log.val "LegalHold.removeSettings")
   void $ permissionCheck ChangeLegalHoldTeamSettings zusrMembership
   ensureReAuthorised zusr mPassword
-  membs <- Data.teamMembersUnsafeForLargeTeams tid
-  removeSettings' tid (Just membs)
+  removeSettings' tid
 
 -- | Remove legal hold settings from team; also disabling for all users and removing LH devices
 removeSettings' ::
   TeamId ->
-  -- | If you've already got the team members you can pass them in otherwise they'll be looked up.
-  Maybe [TeamMember] ->
   Galley ()
-removeSettings' tid mMembers = do
-  membs <- maybe (Data.teamMembersUnsafeForLargeTeams tid) pure mMembers
-  let zothers = map (view userId) membs
-  Log.debug $
-    Log.field "targets" (toByteString . show $ toByteString <$> zothers)
-      . Log.field "action" (Log.val "LegalHold.removeSettings'")
-  let lhMembers = filter ((== UserLegalHoldEnabled) . view legalHoldStatus) membs
-  -- I picked this number by fair dice roll, feel free to change it :P
-  pooledMapConcurrentlyN_ 6 removeLHForUser lhMembers
+removeSettings' tid = do
+  -- Loop through team members and run this action
+  Data.withTeamMembersWithChunks tid action
```

Comment:
Yes, this is potentially true... I think this is better done through the async queue, with a 202. I think that, for now, I think we can simply disable using legalhold for large installations? Let me check something...

Reply:
>  I think we can simply disable using legalhold for large installations? 

We will do that now.

Reply:
Actually this won't work. Let's discuss again tomorrow.

Reply:
I am a little split on if we should use `teamMembersForFanout` here, or fail to remove settings. WDYT?

Reply:
Same here :) But considering that:
 * we should **not** end up in that situation since no code path can lead to it
 * if anyone **does** end up in such a situation, I'd rather try to go through with it given LH

I'd prefer to keep it as is. Assuming retries, etc., following attempts would result in a smaller amount of users being tested.